core/src/replay_stage.rs

```diff
@@ -1090,20 +1090,27 @@ impl ReplayStage {
         progress: &mut ProgressMap,
     ) {
         let start_slot = poh_recorder.read().unwrap().start_slot();
-
-        if let (false, Some(latest_leader_slot)) =
-            progress.get_leader_propagation_slot_must_exist(start_slot)
```

Comment:
I believe in the old code `_must_exist` was a sanity check similar to here:
https://github.com/solana-labs/solana/blob/2b5e00d36/core/src/progress_map.rs#L492

So far the only scenario which we have seen this to fail was a consensus bug resulting in a hash mismatch.
i.e the sanity check here was not the issue but the culprit was somewhere else; and in fact having the check here was useful because that consensus bug was irrecoverable anyways, and helped to identify which slot exactly had diverged.

So if there is a legit scenario (i.e. one without a consensus/runtime bug) which can cause this to panic, then we need to account for that and not always panic here. Otherwise I think it is good to keep the sanity check to avoid introducing consensus bugs. Silencing the panic would only make it harder to identify those bugs early on.